src/connectors/tinyman/tinyman.controllers.ts

```diff
@@ -0,0 +1,160 @@
+      );
+    } else {
+      throw new HttpException(
+        500,
+        UNKNOWN_ERROR_MESSAGE,
+        UNKNOWN_ERROR_ERROR_CODE
+      );
+    }
+  }
+
+  return {
+    network: algorand.network,
+    timestamp: startTimestamp,
+    latency: latency(startTimestamp, Date.now()),
+    base: req.base,
+    quote: req.quote,
+    amount: req.amount,
+    rawAmount: req.amount,
+    expectedAmount: String(trade.expectedAmount),
+    price: String(trade.expectedPrice),
+    gasPrice: 0,
+    gasPriceToken: algorand.nativeTokenSymbol,
+    gasLimit: 0,
+    gasCost: '0.001',
+  };
+}
+
+export async function trade(
+  algorand: Algorand,
+  tinyman: Tinyman,
+  req: TradeRequest
+): Promise<TradeResponse> {
+  const startTimestamp: number = Date.now();
+
+  const limitPrice = req.limitPrice;
+  const account: Account = await algorand.getAccountFromAddress(req.address);
+
+  let trade;
+  try {
+    trade = await tinyman.estimateTrade(<PriceRequest>req);
+  } catch (e) {
+    throw new HttpException(
+      500,
+      TRADE_FAILED_ERROR_MESSAGE,
+      TRADE_FAILED_ERROR_CODE
+    );
+  }
+
+  const estimatedPrice = trade.expectedPrice;
+  logger.info(
+    `Expected execution price is ${estimatedPrice}, ` +
+      `limit price is ${limitPrice}.`
+  );
+
+  if (req.side === 'BUY') {
+    if (limitPrice && new Decimal(estimatedPrice).gt(new Decimal(limitPrice))) {
+      logger.error('Swap price exceeded limit price.');
+      throw new HttpException(
+        500,
+        SWAP_PRICE_EXCEEDS_LIMIT_PRICE_ERROR_MESSAGE(
+          estimatedPrice,
+          limitPrice
+        ),
+        SWAP_PRICE_EXCEEDS_LIMIT_PRICE_ERROR_CODE
+      );
+    }
+  } else {
+    if (limitPrice && new Decimal(estimatedPrice).lt(new Decimal(limitPrice))) {
+      logger.error('Swap price lower than limit price.');
+      throw new HttpException(
+        500,
+        SWAP_PRICE_LOWER_THAN_LIMIT_PRICE_ERROR_MESSAGE(
+          estimatedPrice,
+          limitPrice
+        ),
+        SWAP_PRICE_LOWER_THAN_LIMIT_PRICE_ERROR_CODE
+      );
+    }
+  }
+  const tx = await tinyman.executeTrade(
+    account,
+    trade.trade,
+    req.side === 'BUY'
+  );
+
+  logger.info(`${req.side} swap has been executed.`);
+
+  return {
+    network: algorand.network,
+    timestamp: startTimestamp,
+    latency: latency(startTimestamp, Date.now()),
+    base: req.base,
+    quote: req.quote,
+    amount: req.amount,
+    rawAmount: req.amount,
+    expectedIn: String(trade.expectedAmount),
+    price: String(estimatedPrice),
+    gasPrice: 0,
+    gasPriceToken: algorand.nativeTokenSymbol,
+    gasLimit: 0,
+    gasCost: '0.001',
+    txHash: tx.txnID,
+  };
+}
+
+export async function estimateGas(
+  algorand: Algorand,
+  _tinyman: Tinyman
+): Promise<EstimateGasResponse> {
+  return {
+    network: algorand.network,
+    timestamp: Date.now(),
+    gasPrice: 0,
+    gasPriceToken: algorand.nativeTokenSymbol,
+    gasLimit: 0,
+    gasCost: '0.001',
```

Comment:
Same here.  Since we have access to `algorand` in both cases, I think this value should be supplied by the chain and ultimately come from the config (since we're pinning it to a specific value).